lightning/src/ln/priv_short_conf_tests.rs

```diff
@@ -0,0 +1,286 @@
+		assert_eq!(*node_id, nodes[1].node.get_our_node_id());
+		msg.clone()
+	} else { panic!("Unexpected event"); };
+	if let MessageSendEvent::SendChannelUpdate { ref node_id, msg: _ } = as_msg_events[1] {
+		assert_eq!(*node_id, nodes[1].node.get_our_node_id());
+	} else { panic!("Unexpected event"); }
+
+	nodes[1].node.handle_funding_locked(&nodes[0].node.get_our_node_id(), &as_funding_locked);
+	let bs_msg_events = nodes[1].node.get_and_clear_pending_msg_events();
+	assert_eq!(bs_msg_events.len(), 1);
+	if let MessageSendEvent::SendChannelUpdate { ref node_id, msg: _ } = bs_msg_events[0] {
+		assert_eq!(*node_id, nodes[0].node.get_our_node_id());
+	} else { panic!("Unexpected event"); }
+
+	send_payment(&nodes[0], &[&nodes[1]], 100_000);
+
+	// After 6 confirmations, as required by the spec, we'll send announcement_signatures and
+	// broadcast the channel_announcement (but not before exactly 6 confirmations).
+	connect_blocks(&nodes[0], 4);
+	assert!(nodes[0].node.get_and_clear_pending_msg_events().is_empty());
+	connect_blocks(&nodes[0], 1);
+	nodes[1].node.handle_announcement_signatures(&nodes[0].node.get_our_node_id(), &get_event_msg!(nodes[0], MessageSendEvent::SendAnnouncementSignatures, nodes[1].node.get_our_node_id()));
+	assert!(nodes[1].node.get_and_clear_pending_msg_events().is_empty());
+
+	connect_blocks(&nodes[1], 5);
+	let bs_announce_events = nodes[1].node.get_and_clear_pending_msg_events();
+	assert_eq!(bs_announce_events.len(), 2);
+	let bs_announcement_sigs = if let MessageSendEvent::SendAnnouncementSignatures { ref node_id, ref msg } = bs_announce_events[0] {
+		assert_eq!(*node_id, nodes[0].node.get_our_node_id());
+		msg.clone()
+	} else { panic!("Unexpected event"); };
+	let (bs_announcement, bs_update) = if let MessageSendEvent::BroadcastChannelAnnouncement { ref msg, ref update_msg } = bs_announce_events[1] {
+		(msg.clone(), update_msg.clone())
+	} else { panic!("Unexpected event"); };
+
+	nodes[0].node.handle_announcement_signatures(&nodes[1].node.get_our_node_id(), &bs_announcement_sigs);
+	let as_announce_events = nodes[0].node.get_and_clear_pending_msg_events();
+	assert_eq!(as_announce_events.len(), 1);
+	let (announcement, as_update) = if let MessageSendEvent::BroadcastChannelAnnouncement { ref msg, ref update_msg } = as_announce_events[0] {
+		(msg.clone(), update_msg.clone())
+	} else { panic!("Unexpected event"); };
+	assert_eq!(announcement, bs_announcement);
+
+	for node in nodes {
+		assert!(node.net_graph_msg_handler.handle_channel_announcement(&announcement).unwrap());
+		node.net_graph_msg_handler.handle_channel_update(&as_update).unwrap();
+		node.net_graph_msg_handler.handle_channel_update(&bs_update).unwrap();
+	}
+}
+#[test]
+fn test_1_conf_open() {
+	do_test_1_conf_open(ConnectStyle::BestBlockFirst);
+	do_test_1_conf_open(ConnectStyle::TransactionsFirst);
+	do_test_1_conf_open(ConnectStyle::FullBlockViaListen);
+}
+
+#[test]
+fn test_routed_scid_alias() {
```

Comment:
Could we move the first payment's part of this test into `lightning-invoice/utils` to get some coverage on the invoice util changes too?

Reply:
Hmmm, the change there is trivial, how about I just add an assertion to one of the tests in `utils` there to ensure the invoice is generated correctly?